docs/docs/concepts/volumes.md

```diff
@@ -112,19 +121,99 @@ $ dstack delete -f vol.dstack.yaml
 If the volume was created using `dstack`, it will be physically destroyed along with the data.
 If you've registered an existing volume, it will be de-registered with `dstack` but will keep the data.
 
+## Instance volumes
+
+> Instance volumes are currently supported on all backends except `runpod`, `vastai` and `kubernetes`.
+
+Unlike [network volumes](#network-volumes), which are persistent external resources mounted over network,
+instance volumes are part of the instance storage. Basically, the instance volume is a filesystem path
+(a directory or a file) mounted inside the run container.
+
+As a consequence, the contents of the instance volume are specific to the instance
+where the run is executed, and data persistence, integrity, and even existence are guaranteed only if the subsequent run
+is executed on the same exact instance, and there is no other runs in between.
+
+### Manage volumes { #manage-instance-volumes }
+
+You don't need to create or delete instance volumes, and they are not displayed in the
+[`dstack volume list`](../reference/cli/index.md#dstack-volume-list) command output.
+
+### Attach a volume { #attach-instance-volume }
+
+Dev environments, tasks, and services let you attach any number of instance volumes.
+To attach an instance volume, specify the `instance_path` and `path` in the `volumes` property:
+
+<div editor-title=".dstack.yml">
+
+```yaml
+type: dev-environment
+# A name of the dev environment
+name: vscode-vol
+
+ide: vscode
+
+# Map the instance path to any container path
+volumes:
+  - instance_path: /mnt/volume
```

Comment:
Unsure about `instance_path` and `path`.
Perhaps `instance_path` and `container_path` may be more consistent.
Also, I personally suggest promoting everywhere the short syntax:

```
volumes:
  - <instance path>:<container path>
 ```